### ISSUE TITLE
GitHub Desktop clones the wrong repository

### ISSUE THREAD

**Issue by literallysofia** [desktop/desktop#3969]
### Description

When I clone a certain private repository I own, in this case, feup-lbaw, while cloning GitHub changes the path, and starts cloning feup-laig, another public repository I own. That said, it states that feup-lbaw is feup-laig, so basically GitHub Desktop clones the wrong repository.
I tried cloning using the terminal, and it works, but when I add it to Github Desktop, it changes back to the wrong repository again...
I checked the database and it still contains old info about repositories I once deleted. I would like to clear its cache as well but I don't know how. I tried uninstalling but it didn't work.

### Version

**GitHub Desktop version:** 1.0.13

**OS version:** macOS High Sierra Version 10.13.2

### Steps to Reproduce

<img width="448" alt="screen shot 2018-02-08 at 11 06 22" src="https://user-images.githubusercontent.com/19331863/35969854-22a947f4-0cc0-11e8-8c5e-5fbecb9c1ce4.png">
<img width="1440" alt="screen shot 2018-02-08 at 10 13 37" src="https://user-images.githubusercontent.com/19331863/35969861-27b07da8-0cc0-11e8-90f2-ef8c00365bb9.png">
<img width="1440" alt="screen shot 2018-02-08 at 10 13 45" src="https://user-images.githubusercontent.com/19331863/35969867-2a992a2e-0cc0-11e8-9be0-61ec92d49035.png">
<img width="768" alt="screen shot 2018-02-08 at 10 13 52" src="https://user-images.githubusercontent.com/19331863/35969869-2db1c50e-0cc0-11e8-8c49-01fd3e43c609.png">
<img width="769" alt="screen shot 2018-02-08 at 10 13 59" src="https://user-images.githubusercontent.com/19331863/35969875-30eaad62-0cc0-11e8-848a-9d68d02e3d83.png">
<img width="1013" alt="screen shot 2018-02-08 at 10 17 42" src="https://user-images.githubusercontent.com/19331863/35969877-321ae79c-0cc0-11e8-8239-523cb303e89b.png">

As you can see, the URL is wrong in the database.

**Expected behavior:** Clone the repository feup-lbaw

**Actual behavior:** Clones the repository feup-laig, but it still names the remote folder "feup-lbaw", but its content is completely wrong.

**Reproduces how often:** It only happens when trying to clone this specific repository.

**Comment by tierninho** [comment]
@literallysofia thanks for filing this. As we investigate, is it possible to attach the logs? `Help > Show User Logs`

Also, to clear the cache, open the console with `View > Toggle Developer Tools` and follow the steps here:

![screen shot 2018-02-08 at 10 07 03 am](https://user-images.githubusercontent.com/14828183/35990101-e0bd057c-0cb7-11e8-8ede-d2f54fb81532.png)

**Comment by literallysofia** [comment]
@tierninho Thanks for replying! Unfortunately, clearing the cache didn't work. I'll attach the logs.
[2018-02-08.desktop.production.log](https://github.com/desktop/desktop/files/1708780/2018-02-08.desktop.production.log)

**Comment by iAmWillShepherd** [comment]
@literallysofia, In your steps to reproduce, I noticed you manually entered the URL. Can you try cloning from the **GitHub.com** tab to rule out any potential errors?

After re-cloning using the method described above could you run the following command inside the **feup-lbaw** repo?
`git remote -vv`

**Comment by literallysofia** [comment]
@iAmWillShepherd It doesn't matter which way I clone, I tried it both ways. Same thing happens.
`git remote -vv` shows the wrong path, this one: https://github.com/literallysofia/feup-laig.git
but I want https://github.com/literallysofia/feup-lbaw.git

**Comment by iAmWillShepherd** [comment]
Hmm, can you clone the repo in question with the CLI and run the command, `git remote -vv`, and paste its output?

**Comment by literallysofia** [comment]
@iAmWillShepherd As I mentioned above, it works just fine with the CLI, but I was hoping to fix the problem with GitHub Desktop. 
<img width="568" alt="screen shot 2018-02-08 at 21 53 34" src="https://user-images.githubusercontent.com/19331863/36000375-a3261934-0d1a-11e8-82d4-51ca24e93c5b.png">

**Comment by iAmWillShepherd** [comment]
@literallysofia I understand, but I need reproduction steps I can use to better help. Since your repository is private, I cannot do these actions myself to diagnose the problem.

**Comment by iAmWillShepherd** [comment]
@literallysofia are you be able to add me as a collaborator so I can figure out why this particular repo is causing problems for the client?

**Comment by literallysofia** [comment]
@iAmWillShepherd This issue doesn't happen on windows, only mac. I'll make it public.

**Comment by iAmWillShepherd** [comment]
@literallysofia that's even weirder! 😕

**Comment by iAmWillShepherd** [comment]
Tried to reproduce on macOS.

The _clone URL_ is `literallysofia/feup-lbaw.git` as expected

![2018-02-08_16-59-32 894](https://user-images.githubusercontent.com/1715082/36003049-04469f00-0cf2-11e8-8d1b-8dd93911aee7.png)

The correct _clone URL_ and _name_ is inserted correctly as `literallysofia/feup-lbaw.git` and `feup-lbaw` as expected

![2018-02-08_17-03-10 643](https://user-images.githubusercontent.com/1715082/36003083-1b55917e-0cf2-11e8-9115-0f82f234e820.png)

**Comment by literallysofia** [comment]
@iAmWillShepherd Thanks for checking, I don't know what to do then.

**Comment by iAmWillShepherd** [comment]
I just looked at your repository's git database and everything looks normal. Can you try this?

1. Deleting all databases
1. Clear local storage
1. Restart the application

**Comment by literallysofia** [comment]
I managed to fix the issue!
Install [AppCleaner](https://freemacsoft.net/appcleaner/) and run `sudo rm -f /usr/local/bin/github`, works like a dream!

**Comment by Wicko-Design** [comment]
I have a project I have started on my local drive. But when I select it to upload as new repository to GitHub it create a repository that is the same as another repository I already have and then comes up with an error saying it can't find it. I then have to remove it from GitHub Desktop. So stuck adding a new repository to github using the desktop app. Currently using version 1.5.0 on mac running Mojave

**Comment by abegehr** [comment]
I'm still having the problem even a

> Something like this happens to me too.
> 
> * I had one repository that I renamed (SpendCA -> SpendCA.Mobile).
> * Create another repository with name (SpendCA)
> 
> Now, when I clone this new repository, it appears the name of the old one.

Same here!
The fix from @literallysofia did not work for me. – What should I do with AppCleaner? Delete and reinstall Github Desktop?
Cloning through the command line works as expected. I'll go that route for the time being.

**Comment by abegehr** [comment]
It is working now after deleting GitHub Desktop with AppCleaner, running the 'rm' from @literallysofia's post, and reinstalling.

**Comment by ternava** [comment]
Why this is closed? It's something that still keeps appearing as an issue, even in Windows. Does anyone know why exactly this is happening?

**Comment by steveward** [comment]
@ternava could you reproduce this issue and share the log files from GitHub Desktop? To access the log files go to the file menu in GitHub Desktop and select `Help` > `Show Logs in Finder (macOS) or Explorer (Windows)`.

**Comment by samuelsweet** [comment]
After struggling with this issue on Windows for a couple of hours now (and trying everything under the sun), the way I finally resolved was by uninstalling... deleting the "%appdata%/Github Desktop" folder. Then reinstall and it finally works correctly.

**Comment by rbezantTC** [comment]
This happened for me as well. Github Desktop, Windows 7.

I don't recall the exact steps but it occurred after I had created a new repo and gave it the same name as an old repo after I had renamed the old repo to something else.

i.e. "repo" named to "old repo". New repo created, named as "repo"

It appears to be a caching issue. Fix by:

- Closing Github Desktop
- Deleting the content in your _HDD:\Users\\[username]\AppData\Roaming\Github Desktop\Cache_ folder
- Re-open Github Desktop and re-clone
